spec.bs

```diff
@@ -1356,6 +1467,204 @@ To <dfn>build an interest group passed to generateBid</dfn> given an [=interest
+[=auction config=]-or-null |topLevelAuctionConfig|, and a new [=map=] |negativeInfo| whose
+[=map/keys=] are [=tuples=] consisting of an [=origin=] and a [=string=], and whose [=map/values=]
+are [=tuples=] consisting of an [=origin=] and a [=byte sequence=]:
+
+  1. Let |auctionNonce| be |auctionConfig|'s [=auction config/auction nonce=].
+  1. Let |capturedAdditionalBidsHeaders| be [=this=]'s [=relevant global object=]'s
+    [=associated Document's=] [=node navigable's=] [=traversable navigable's=]
+    [=captured additional bids headers=].
+  1. Let |additionalBids| be a new [=list=] of [=additional bids=].
+  1. [=list/For each] |encodedAdditionalBid| of |capturedAdditionalBidsHeaders|[|auctionNonce|]:
+    1. Let |decodedSignedAdditionalBid| be the result of running [=forgiving-base64 decode=] with
+      |encodedAdditionalBid|.
+    1. If |decodedSignedAdditionalBid| is failure, [=iteration/continue=].
+    1. Increment |auctionConfig|’s [=auction config/scoring dependency count=].
+    1. Let |parsedSignedAdditionalBid| be the result of [=parse a JSON string to an infra value=]
+      with |decodedSignedAdditionalBid|.
+    1. Decrement |auctionConfig|’s [=auction config/scoring dependency count=], and
+      [=iteration/continue=] if any of the following conditions hold:
+      * |parsedSignedAdditionalBid| is not a [=map=];
+      * |parsedSignedAdditionalBid|["bid"] does not [=map/exist=], or it's not a [=string=];
+      * |parsedSignedAdditionalBid|["signatures"] does not [=map/exist=], or it's not a [=list=].
+    1. Let |signatures| be a new [=list=] of [=signed additional bid signatures=].
+    1. Let |decodeSignatureFailed| be false.
+    1. [=list/For each=] |sig| of |parsedSignedAdditionalBid|["signatures"]:
+      1. Set |decodeSignatureFailed| to true and [=iteration/break=] if any of the following
+        conditions hold:
+        * |sig| is not a [=map=];
+        * |sig|["key"] does not [=map/exist=], or it's not a [=string=];
+        * |sig|["signature"] does not [=map/exist=], or it's not a [=string=].
+      1. Let |maybeKey| be the result of running [=forgiving-base64 decode=] with |sig|["key"].
+      1. Let |maybesignature| be the result of running [=forgiving-base64 decode=] with
+        |sig|["signature"].
+      1. Set |decodeSignatureFailed| to true and [=iteration/break=] if any of the following
+        conditions hold:
+        * |maybeKey| is failure, or its [=byte sequence/length=] is not 32;
+        * |maybesignature| is failure, or its [=byte sequence/length=] is not 64;
+      1. Let |signature| be a [=signed additional bid signatures=], whose
+        [=signed additional bid signature/key=] is |maybeKey|, and
+        [=signed additional bid signature/signature=] is |maybesignature|.
+      1. [=list/Append=] |signature| to |signatures|.
+    1. If |decodeSignatureFailed| is true, then decrement |auctionConfig|’s
+      [=auction config/scoring dependency count=], and [=iteration/continue=].

```

Comment:
I feel like the number of times you are doing it suggests you should just have a routine that returns failure called within the loop on line 1481, so you can check failure + continue once rather than repeating it all over the place